mslice/models/workspacemanager/file_io.py

```diff
@@ -0,0 +1,143 @@
+from __future__ import (absolute_import, division, print_function)
+import os.path
+from mantid.api import IMDHistoWorkspace
+from mantid.simpleapi import CreateMDHistoWorkspace, SaveMD, SaveNexus, SaveAscii
+from mslice.util.qt.QtWidgets import QFileDialog
+
+import numpy as np
+from scipy.io import savemat
+
+
+def get_save_directory(multiple_files=False, save_as_image=False, default_ext=None):
+    '''
+    Show file dialog so user can choose where to save.
+    :param multiple_files: boolean - whether more than one file is being saved
+    :param save_as_image: boolean - whether to allow saving as a .png/pdf
+    :param default_ext: file extension that is selected by default
+    :return: path to save directory, name to save the file as, file format extension
+    :raises: RuntimeError if dialog is cancelled
+    '''
+    if multiple_files:
+        return QFileDialog.getExistingDirectory(), None, default_ext
+    else:
+        file_dialog = QFileDialog()
+        file_dialog.setAcceptMode(QFileDialog.AcceptSave)
+
+        filter = "Nexus (*.nxs);; Ascii (*.txt);; Matlab (*.mat)"
+        if save_as_image:
+            filter += ";; Image (*.png);; PDF (*.pdf)"
+        file_dialog.setNameFilter(filter)
+
+        if default_ext:
+            ext_to_qtfilter = {'.nxs': 'Nexus (*.nxs)', '.txt': 'Ascii (*.txt)', '.mat': 'Matlab (*.mat)', }
+            file_dialog.selectNameFilter(ext_to_qtfilter[default_ext])
+        if (file_dialog.exec_()):
+            path = file_dialog.selectedFiles()[0] + file_dialog.selectedFilter()[-5:-1]
```

Comment:
Could you add a check here if the file name already has an extension? (If I type something like `test.nxs` in the dialog, I get a file named `test.nxs.nxs`)